src/testing/govulncheck/govulncheck.go

```diff
@@ -0,0 +1,49 @@
+package govulncheck
+
+import (
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+
+	"github.com/bazelbuild/rules_go/go/runfiles"
+)
+
+// SetupEnv sets up the environment for govulncheck (which shells out to Go), and returns the
+// working directory for the _main module.
+func SetupEnv() string {
+	// Find the go Bazel uses and put it in $PATH.
+	if goBin, err := runfiles.Rlocation("rules_go~~go_sdk~go_sdk/bin/go"); err != nil {
+		fmt.Fprintf(os.Stderr, "can't find go binary; using installed go instead: %v\n", err)
+	} else {
+		path := os.Getenv("PATH")
+		path = filepath.Dir(goBin) + ":" + path
+		os.Setenv("PATH", path)
+	}
+
+	// Use a common cache between tests and the //tools/govulncheck command.  We avoid
+	// $XDG_CONFIG_CACHE because that's the cache for the "host" go and govulncheck.  This might
+	// be a different version of Go than what the user has installed.
```

Comment:
Does Bazel provide a standard cache location?

Reply:
For stuff it understands only.  We have to run outside of the sandbox so that it can see the entire repo.  (Bazel can make a binary into a GOPATH for that binary, which would be nice here, but govulncheck doesn't understand GOPATH style projects.)